[REFACTOR] 단위 테스트 코드 리팩터링

jye326 opened the pull request:
## Issue Number
closed #855 

## As-Is
<!-- 문제 상황 정의 -->

- 엔티티 간 연관 관계로 인해 Service 레이어의 단위 테스트에 베이스 엔티티를 생성하야만 합니다.
- 이로 인해 테스트 코드에 불필요한 중복이 늘어나고 가독성이 저하되었습니다.

## To-Be
<!-- 변경 사항 -->
### FixtureUtil 클래스 추가
- 기본/특수 설정 값을 가지는 베이스 엔티티 (Member, Mentoring, Certificate, Reservation) 생성을 돕는 유틸 클래스입니다.

### Service 단위 테스트 리팩터링
- FixtureUtil 클래스를 활용하여 Service 레이어의 단위 테스트를 리팩터링 하여 가독성을 높였습니다!

## Check List
- [x] 테스트가 전부 통과되었나요?
- [x] 모든 commit이 push 되었나요?
- [x] merge할 branch를 확인했나요?
- [x] Assignee를 지정했나요?
- [x] Label을 지정했나요?
- [x] 닫을 이슈 번호를 지정했나요?


## (Optional) Additional Description
- 눈알이 빠지는 줄 알았습니다.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- Tests
  - 공통 테스트 픽스처 유틸을 도입해 회원·멘토링·예약·리뷰 시나리오를 일관된 방식으로 생성하도록 개선했습니다.
  - 중복 설정을 제거하고 어서션을 정리하여 가독성과 안정성을 높였습니다.
- Refactor
  - 수동 객체 생성과 설정을 픽스처 기반으로 전환하고 테스트 흐름을 간소화했습니다.
- Chores
  - 테스트 임포트와 표현을 정리하고 문구를 일부 다듬어 유지보수성을 향상했습니다.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jye326 committed:
refactor: AuthServiceTest 코드 중복 제거를 위한 newMember 메서드 분리

- 테스트 코드 내 Member 객체 생성 로직을 newMember 메서드로 통합
- 코드 중복 제거로 가독성 및 유지보수성 향상

jye326 committed:
refactor: MemberServiceTest 코드 중복 제거를 위한 getTestMember, getTestMentoring 메서드 분리

- 테스트 코드 내 Member, Mentoring 객체 생성 로직을 메서드로 통합
- 코드 중복 제거로 가독성 및 유지보수성 향상

jye326 committed:
refactor: CertificateServiceTest 코드 중복 제거를 위한 getTestMember, getTestMentoring 메서드 분리

- 테스트 코드 내 Member, Mentoring, Certificate 객체 생성 로직을 메서드로 통합
- 코드 중복 제거로 가독성 및 유지보수성 향상

jye326 committed:
refactor: 테스트 코드 객체 생성 로직 통합 및 중복 제거

- FixtureUtil 클래스를 생성하여 AuthServiceTest, CertificateServiceTest, MemberServiceTest 등 각 테스트의 중복된 객체 생성 로직 통합
- 코드 중복 제거로 가독성 및 유지보수성 개선

jye326 committed:
refactor: MentoringServiceTest 코드 중복 제거를 위한 FixtureUtil 활용

- 테스트 코드 내 Member, Mentoring 객체 생성 로직 FixtureUtil로 통합
- 불필요한 엔티티 생성 코드 제거로 가독성 및 유지보수성 개선

jye326 committed:
refactor: FixtureUtil 메서드 이름 및 사용 개선으로 테스트 코드 가독성 향상

- getTestMentee, getTestMentor 메서드 분리로 역할 구분 명확화
- 기존 getTestMember 호출 부분 getTestMentee로 교체
- 불필요한 객체 생성 및 import 문 제거로 코드 간소화 및 일관성 유지

jye326 committed:
refactor: ReservationServiceTest에 FixtureUtil 적용으로 테스트 코드 간소화 및 중복 제거

- FixtureUtil 활용하여 ReservationServiceTest 내 Member, Mentoring, Reservation 객체 생성 통합
- 중복된 엔티티 생성 코드 제거로 가독성, 유지보수성 및 일관성 개선

jye326 committed:
refactor: ReviewRepositoryTest에 FixtureUtil 적용으로 테스트 코드 간소화 및 중복 제거

- FixtureUtil 활용하여 Member, Mentoring, Reservation, Review 객체 생성 통합
- 불필요한 엔티티 생성 및 import 문 제거로 코드 간소화 및 일관성 유지
- 중복된 테스트 코드 제거로 가독성 및 유지보수성 개선

jye326 committed:
refactor: AuthServiceTest 및 ReservationServiceTest 코드 개선을 위한 FixtureUtil 메서드 수정 및 통합

- 기존 loginId 값을 menteeId로 변경하여 테스트 가독성 및 통일성 향상
- FixtureUtil에 getTestCompletedReservation 메서드 추가 및 기존 getTestReservation을 getTestPendingReservation으로 변경
- ReservationServiceTest 내 Reservation 관련 생성 로직을 FixtureUtil 메서드 호출로 통합
- 불필요한 import 제거 및 코드 간소화로 유지보수성 개선

jye326 committed:
refactor: ReviewServiceTest에 FixtureUtil 적용을 통한 코드 간소화 및 중복 제거

- FixtureUtil 활용으로 Member, Mentoring, Reservation, Review 객체 생성 통합
- 중복된 엔티티 생성 코드 제거 및 불필요한 import 정리
- 테스트 코드 가독성과 유지보수성 개선.

jye326 committed:
refactor: ReviewRepositoryTest에 FixtureUtil 변경 사항 반영 및 테스트 코드 개선

- getTestReservation 메서드를 getTestPendingReservation으로 대체하여 코드 일관성 유지
- 테스트 코드 간소화 및 가독성 향상

jye326 committed:
refactor: MentoringServiceTest에 FixtureUtil 변경 사항 반영 및 테스트 코드 개선

- FixtureUtil 메서드(getTestCompletedReservation, getTestPendingReservation, getTestReview) 적용으로 코드 간소화
- 중복 객체 생성 로직 제거 및 일관성 유지
- 테스트 코드 가독성과 유지보수성 향상

jye326 commented:
@coderabbitai 놓친 거 있나 한번 싹 체크해봐

jye326 commented on backend/fittoring/src/test/java/fittoring/application/FixtureUtil.java:
```diff
@@ -0,0 +1,90 @@
+package fittoring.application;
+
+import fittoring.domain.model.*;
```
싫어

jye326 commented on backend/fittoring/src/test/java/fittoring/application/member/service/MemberServiceTest.java:
```diff
@@ -58,98 +57,54 @@ void setUp() {
+            softAssertions.assertThat(memberInfo.gender()).isEqualTo(member.getGender());
+            softAssertions.assertThat(memberInfo.phoneNumber()).isEqualTo(member.getPhoneNumber());
         });
     }
```
좋아.

jye326 commented on backend/fittoring/src/test/java/fittoring/application/FixtureUtil.java:
```diff
@@ -0,0 +1,90 @@
+        );
+    }
+
+    public static Mentoring getTestMentoring(Member member) {
+        member.registerAsMentor();
+        return new Mentoring(
+                member,
+                5000,
+                5,
+                "content",
+                "introduction",
+                "https://chatRoomUrl"
+        );
+    }
```
아니, 이게 맞아

jye326 committed:
refactor: MemberServiceTest 및 ReviewServiceTest에 FixtureUtil 변경 사항 반영

- getTestMentor, getTestMentee 메서드 활용으로 객체 생성 코드 통일
- Mentor, Mentee 간 역할 구분 명확화 및 테스트 코드 간소화
- 코드 가독성과 유지보수성 개선

jye326 committed:
refactor: ReservationServiceTest에 FixtureUtil 변경 사항 반영 및 테스트 코드 간소화

- getTestPendingReservation 메서드 활용으로 Reservation 객체 생성 코드 통합
- 중복된 엔티티 생성 로직 제거로 가독성 및 유지보수성 향상

jye326 commented on backend/fittoring/src/test/java/fittoring/application/reservation/service/ReservationServiceTest.java:
```diff
@@ -168,103 +143,57 @@ void createReservationFail1() {
-        entityManager.persist(reservation3);
+        entityManager.persist(new Reservation("content", Status.PENDING, mentoring, mentee1));
+        entityManager.persist(new Reservation("content", Status.PENDING, mentoring, mentee2));
+        entityManager.persist(new Reservation("content", Status.PENDING, mentoring, mentee3));
```
좋다

jye326 committed:
refactor: ReviewServiceTest에 FixtureUtil 변경 사항 반영 및 테스트 코드 개선

- getTestMentor 메서드 적용으로 Mentor 객체 생성 코드 통일
- 중복된 엔티티 생성 코드 제거로 가독성과 유지보수성 향상

kimdevv commented on backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java:
```diff
@@ -88,16 +88,9 @@ void register() {
-                Password.from("password")
-        );
-        em.persist(member);
+        String loginId = "menteeId";
```
p3: `FixtureUtil.getTestMentee().getLoginId()`로 `loginId` 변수를 할당하면 픽스처가 변경되어도 테스트는 변경되지 않아도 될 것 같아요!

kimdevv commented on backend/fittoring/src/test/java/fittoring/application/auth/service/AuthServiceTest.java:
```diff
@@ -175,16 +147,9 @@ void login2() {
+        Member savedMember = em.persist(FixtureUtil.getTestMentee());
 
-        String loginId = "loginId";
+        String loginId = "menteeId";
```
위와 동일한 의견입니다~

kimdevv reviewed (approved):
Nice work! - Our test codes looks much simpler now! 번거로운 작업이었을 텐데 고생하셨습니다.
I also read [the post](https://jojoldu.tistory.com/611) you shared, and it was so helpful. I didn't know about test fixture before, but now I have much better understanding.  I approve. Please check some comments I left.

ppparkta commented on backend/fittoring/src/test/java/fittoring/application/FixtureUtil.java:
```diff
@@ -0,0 +1,90 @@
+import fittoring.domain.model.password.Password;
+
+public class FixtureUtil {
+    public static Member getTestMentee() {
```
p3: 개행 부탁드려요

ppparkta reviewed (approved):
> 테스트 메소드는 그 하나로 완전한 프로그램이 되어야 한다.

좋은 글 소개해 주셔서 감사해요! 여러 엔티티가 여러 서비스에서 쓰이고 있어서 클래스 외부에 static 메서드로 분리해주셨네요. 변경된 코드를 봤는데 많이 깔끔해졌어요. 좋습니다.

jye326 committed:
fix: 테스트 데이터 및 AuthService 테스트 코드 수정

- `FixtureUtil` 테스트 데이터 수정 및 가독성 개선
- 로그인 ID 및 비밀번호 검증 로직 테스트에서 하드코딩 제거
- `AuthServiceTest` 내 불필요한 변수명 변경 및 명확화

dye0p reviewed (approved):
> 눈알이 빠지는 줄 알았습니다.

인상깊네요..

테스트 픽스처 도입 수고많으셨어요! 앞으로 테스트할때 편하게 할 수 있겠군요. 코멘트는 따로 남길게 없어서 바로 머지하겠습니다!
